### PR TITLE
Mute tests that regularly leak SearchHits

### PR DESCRIPTION
**Pull request by jan-elastic**
- [GitHub Issue](https://github.com/elastic/elasticsearch/issues/104570)
- [Test history](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now)))
- This test is failing 54/2247 during the last week

### CODE REVIEW

File: modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java [hunk 1]
```diff
@@ -122,6 +122,7 @@ public void testXContentRoundtrip() throws IOException {
         }
     }
 
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/104570")
```
Comment (jan-elastic) [inline modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java:125]: Is there a way to only mute for Windows? I think it only happens there?

File: modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java [hunk 2]
```diff
@@ -122,6 +122,7 @@ public void testXContentRoundtrip() throws IOException {
         }
```
Comment (jan-elastic) [inline modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java:122]: Note: the is almost identical to https://github.com/elastic/elasticsearch/pull/104684, which didn't get merged, so I'm uncertain whether I should merge this.

File: modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java (hunk 1 above)
Comment (droberts195) [reply modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java:125]: Yes, like this: https://github.com/elastic/elasticsearch/blob/2bc739875465d2e3048e51a81a5d973537c6c460/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/ForecastIT.java#L225

Comment (jan-elastic) [reply modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RatedRequestsTests.java]: Thanks, fixed

**Review by thecoop — APPROVED** [review]

**Pull request merged**